ddl/ddl_api.go

```diff
@@ -422,13 +374,31 @@ func setCharsetCollationFlenDecimal(tp *types.FieldType, specifiedCollates []str
 	return nil
 }
 
-// outPriKeyConstraint is the primary key constraint out of column definition. such as: create table t1 (id int , age int, primary key(id));
-func buildColumnAndConstraint(ctx sessionctx.Context, offset int,
-	colDef *ast.ColumnDef, outPriKeyConstraint *ast.Constraint, tblCharset, tblCollate, dbCharset, dbCollate string) (*table.Column, []*ast.Constraint, error) {
-	// specifiedCollates refers to collates in colDef.Options, should handle them together.
-	specifiedCollates := extractCollateFromOption(colDef)
+// buildColumnAndConstraint builds table.Column and ast.Constraint from the parameters.
+// outPriKeyConstraint is the primary key constraint out of column definition. For example:
+// `create table t1 (id int , age int, primary key(id));`
+func buildColumnAndConstraint(
+	ctx sessionctx.Context,
+	offset int,
+	colDef *ast.ColumnDef,
+	outPriKeyConstraint *ast.Constraint,
+	tblCharset string,
+	tblCollate string,
+) (*table.Column, []*ast.Constraint, error) {
```

Comment:
Ignore the charset of the `database`? If the table was created in old TiDB version, the `charset` of the table will be empty?

Reply:
Oh, I see it, the charset of the `database` will be resolve in the outside of this function.

Reply:
Yes, the charset of table has been resolved from the database before calling this function.